sdks/python/apache_beam/examples/snippets/snippets.py

```diff
@@ -1181,6 +1189,32 @@ def model_bigqueryio(
       create_disposition=beam.io.BigQueryDisposition.CREATE_IF_NEEDED)
   # [END model_bigqueryio_write]
 
+  # [START model_bigqueryio_write_with_storage_write_api]
+  quotes | beam.io.WriteToBigQuery(
+      table_spec,
+      schema=table_schema,
+      method=beam.io.WriteToBigQuery.Method.STORAGE_WRITE_API)
+  # [END model_bigqueryio_write_with_storage_write_api]
```

Comment:
Thank you for letting me know. I didn't know the limitations!
I have just added these notes.👍

Reply:
Hi, @ahmedabu98 .
I tried to write 'TIMESTAMP' type, but I got an error message
```
java.lang.IllegalArgumentException: Input schema is not assignable to output schema.
Input schema=Fields:
.......
INFO 2023-07-03T09:27:24.015855Z Field{name=ts, description=, type=LOGICAL_TYPE<beam:logical_type:micros_instant:v1>, options={{}}}
.......
, Output schema=Fields:
.......
INFO 2023-07-03T09:27:24.016054Z Field{name=ts, description=, type=DATETIME, options={{}}}
.......
``` 
I looked into beam source codes and found that [input mapping](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/typehints/schemas.py#L36) and [output mapping](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java#L324) are not matched in the [schema validation code](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/providers/BigQueryStorageWriteApiSchemaTransformProvider.java#L414)). I think output mapping should be `FieldType.logicalType(SqlTypes.TIMESTAMP)`.
Is there a way to avoid this error, or is `TIMESTAMP` type not supported yet?

Reply:
I looked into the code again and found [this](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/typehints/schemas.py#L787~L799).
It seems that `beam:logical_type:millis_instant:v1` is translated as `FieldType.DATETIME` in [SchemaTranslation](
https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/SchemaTranslation.java#L427), which is matched with the [output mapping](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java#L324).
To use `MillisInstant`, `LogicalType.register_logical_type(MillisInstant)` should be added to the pipeline code.

**Example**
```
LogicalType.register_logical_type(MillisInstant)
quotes | beam.io.WriteToBigQuery(
    table_spec,
    schema=table_schema,
    write_disposition=beam.io.BigQueryDisposition.WRITE_TRUNCATE,
    create_disposition=beam.io.BigQueryDisposition.CREATE_IF_NEEDED,
    method=beam.io.WriteToBigQuery.Method.STORAGE_WRITE_API,
)
```

Reply:
Thanks for looking into this @HaeSe0ng!

@Abacn do you know if it's sufficient for us to include the line `LogicalType.register_logical_type(MillisInstant)` in the connector? e.g. here: https://github.com/apache/beam/blob/7eaef18bcbfb48fb112884564b1362f638fe5834/sdks/python/apache_beam/io/gcp/bigquery.py#L2193-L2194

Reply:
@JoeCMoore are you getting this error even when setting `LogicalType.register_logical_type(MillisInstant)`?

Reply:
@ahmedabu98 No, this was a workaround for the bug. If you need any more details I'd be happy to help.